programs/ssl/ssl_client2.c

```diff
@@ -922,19 +926,28 @@ static int send_cb( void *ctx, unsigned char const *buf, size_t len )
 }
 
 #if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+typedef struct {
+    unsigned char psk[MBEDTLS_PSK_MAX_LEN];
+    size_t psk_len;
+} psk_entry_t;
+
 static int psk_cb( void *p_psk,  mbedtls_ssl_context *context,
                    const unsigned char *identity, size_t  identity_len) {
-    (void)context;
-    (void)p_psk;
-
-    /* INFO: You can also configure the psk inside this callback. */
+    psk_entry_t *psk_info = (psk_entry_t*)p_psk;
+    int ret = 0;
 
     mbedtls_printf( "\n  . Receieved server identity: %.*s\n",  (int)identity_len, identity);
 
+    ret = mbedtls_ssl_set_hs_psk(context, psk_info->psk, psk_info->psk_len);
```

Comment:
Style: spaces inside parentheses (also the three following lines).

Reply:
fixed